src/swap/useSwapQuote.ts

```diff
@@ -52,10 +52,14 @@ const useSwapQuote = () => {
     }
 
     const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
-    const requestUrl = `${networkConfig.approveSwapUrl}?buyToken=${toToken.address}&sellToken=${
-      fromToken.address
-    }&${swapAmountParam}=${swapAmountInWei.toString().split('.')[0]}&userAddress=${walletAddress}`
-
+    const params = {
+      buyToken: toToken.address,
+      sellToken: fromToken.address,
+      [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+      userAddress: walletAddress,
+    }
+    const queryParams = new URLSearchParams({ ...(params as Record<string, string>) }).toString()
```

Comment:
let's refrain from type casting as much as possible and let TS do work for us....i think the type error here that is silenced by the cast is genuine - `walletAddress` can be null and we don't handle this case. 

not sure how best to handle this, maybe the `refreshQuote` function can do an early return if `walletAddress` is not defined? it should never happen anyway but if we did this then this `as Record<string, string>` can be safely removed.

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2767/commits/3873bbbc5c825bb150c56a25f594f4b4d3025244!